kotlinpoet/src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -186,17 +186,21 @@ public class TypeSpec private constructor(
             codeWriter.emit("constructor")
           }
 
-          it.parameters.emit(codeWriter, forceParensOnEmpty = useKeyword, forceNewLines = true) { param ->
-            val property = constructorProperties[param.name]
-            if (property != null) {
-              property.emit(
-                codeWriter, setOf(PUBLIC), withInitializer = false, inline = true,
-                inlineAnnotations = false
-              )
-              param.emitDefaultValue(codeWriter)
-            } else {
-              param.emit(codeWriter, emitKdoc = true, inlineAnnotations = false)
+          if (it.parameters.isNotEmpty()) {
+            it.parameters.emit(codeWriter, forceParensOnEmpty = useKeyword, forceNewLines = true) { param ->
+              val property = constructorProperties[param.name]
+              if (property != null) {
+                property.emit(
+                  codeWriter, setOf(PUBLIC), withInitializer = false, inline = true,
+                  inlineAnnotations = false
+                )
+                param.emitDefaultValue(codeWriter)
+              } else {
+                param.emit(codeWriter, emitKdoc = true, inlineAnnotations = false)
+              }
             }
+          } else {
+            codeWriter.emit("()")
```

Comment:
`parameters.emit()` already encodes this behavior if `forceParensOnEmpty` is true, so I suggest using that. It also looks like this is the only call site that passes a value for `forceParensOnEmpty` so we can explore deleting that parameter altogether and just making the method always emit brackets.

Reply:
Update with your comments @Egorand 😃